### PR TITLE
Refactor code for enabling MSAL-Broker communication via AccountManager without GET_ACCOUNT permission

### PR DESCRIPTION
**Pull request by rpdome**
Also moving MSAL code that can be shared to common.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java [hunk 1]
```diff
@@ -257,6 +265,44 @@ public BrokerAcquireTokenSilentOperationParameters brokerSilentParametersFromBun
...
+            requestBundle.putString(ENVIRONMENT, parameters.getAccount().getEnvironment());
+            requestBundle.putString(ACCOUNT_HOME_ACCOUNT_ID, parameters.getAccount().getHomeAccountId());
+        }
+
+        return requestBundle;
+    }
+
+    public static Bundle getRequestBundleForGetAccounts(@NonNull final OperationParameters parameters) {
+        final Bundle requestBundle = new Bundle();
+        requestBundle.putString(ACCOUNT_CLIENTID_KEY, parameters.getClientId());
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: Can we rename to `ACCOUNT_CLIENTID_KEY` to `ACCOUNT_CLIENT_ID_KEY`?

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java [hunk 2]
```diff
@@ -257,6 +265,44 @@ public BrokerAcquireTokenSilentOperationParameters brokerSilentParametersFromBun
...
+        requestBundle.putString(ACCOUNT_REDIRECT, parameters.getRedirectUri());
+        //Disable the environment and tenantID. Just return all accounts belong to this clientID.
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: What does disable mean in this context?

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java [hunk 3]
```diff
@@ -1187,6 +1193,29 @@ private AuthenticationConstants() {
         public static final String AZA_SCOPE = "aza";
     }
 
+    /**
+     * Represents Broker operations that should be invoked by Authenticator.java (MSAL-Broker AccountManager flow).
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: Why is this class called `MsalAccountManagerOperation` when the javadoc says that it represents Broker Operations?
May be rename to `BrokerAccountManagerOperation`?

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java (hunk 1 above)
Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: I'll keep it as is right now to keep the amount of change as minimal as possible - MSAL and Broker change is already big enough IMHO.

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java (hunk 2 above)
Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: as synced with @iambmelt  offline, we're not passing them to Broker right now.

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java [hunk 4]
```diff
@@ -257,8 +266,48 @@ public BrokerAcquireTokenSilentOperationParameters brokerSilentParametersFromBun
         return new HashSet<>(Arrays.asList(scopes));
     }
 
+    public static Bundle getSilentBrokerRequestBundle(final AcquireTokenSilentOperationParameters parameters) {
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: nit: lets try to follow the adapter conventions here, any reason why it's static?
also rename to `requestBundleFromSilentParameters` to be consistent and remove `static`

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java [hunk 5]
```diff
@@ -257,8 +266,48 @@ public BrokerAcquireTokenSilentOperationParameters brokerSilentParametersFromBun
...
+        final Bundle requestBundle = new Bundle();
+        final BrokerRequest brokerRequest = msalBrokerRequestAdapter.
+                brokerRequestFromSilentOperationParameters(parameters);
+
+        requestBundle.putString(
+                AuthenticationConstants.Broker.BROKER_REQUEST_V2,
+                new Gson().toJson(brokerRequest, BrokerRequest.class)
+        );
+
+        requestBundle.putInt(
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: not part of this change, but we should revisit why `CALLER_INFO_UID` is not part of `BrokerRequest`, could be a miss on my part when i was implementing this

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java [hunk 6]
```diff
@@ -257,8 +266,48 @@ public BrokerAcquireTokenSilentOperationParameters brokerSilentParametersFromBun
...
+                AuthenticationConstants.Broker.CALLER_INFO_UID,
+                parameters.getAppContext().getApplicationInfo().uid
+        );
+
+        return requestBundle;
+    }
+
+    public static Bundle getRequestBundleForRemoveAccount(@NonNull final OperationParameters parameters) {
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: same here , rename to something like `removeAccountRequestBundleFromParameters`

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java [hunk 7]
```diff
@@ -275,6 +324,20 @@ public static Bundle getBrokerHelloBundle(@NonNull final OperationParameters par
         return requestBundle;
     }
 
+    public static Bundle getRequestBundleForRemoveAccountFromSharedDevice(@NonNull final OperationParameters parameters) {
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: same feedback with `name` and `static` here

File: common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java [hunk 8]
```diff
@@ -422,4 +430,18 @@ public static void verifyRemoveAccountResultFromBundle(@NonNull final Bundle bun
         }.getType();
         return builder.create().fromJson(accountJson, listOfCacheRecords);
     }
+
+    public static AcquireTokenResult getAcquireTokenResult(@NonNull final Bundle resultBundle) throws BaseException {
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java]: same feedback with `name` and `static` here

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java (hunk 4 above)
Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: After taking another look, it makes more sense to put these functions in BrokerBaseStrategy instead.

I'll move them there, and I'll remove static and rename the method for consistency also.

File: common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java (hunk 5 above)
Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: Do you think it makes sense to set CALLER_INFO_UID to every request bundle? 

(Note: this value is already obtainable through Binder.getCallerUid and AccountManager's bundle.)

Comment (kreedula) [reply common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: yeah I think this is not needed as we are explicity adding anyway her e: https://github.com/AzureAD/ad-accounts-for-android/blob/dev/AADAuthenticator/src/msal/java/com/microsoft/identity/client/MicrosoftAuthService.java#L293

Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java]: Keeping it for now because we have a validation logic that check this in Broker.
